File: src/main/java/com/jacuum/web/dto/MapSnapshot.java
package com.jacuum.web.dto;

import java.util.List;

public record MapSnapshot(
    int width,
    int height,
    int startX,
    int startY,
    int totalFloor,
    List<TileSnapshot> tiles
) {
    public record TileSnapshot(
        int x, int y,
        boolean wallNorth, boolean wallSouth, boolean wallEast, boolean wallWest
    ) {}
}

File: src/main/kotlin/com/jacuum/web/dto/MapSnapshot.kt
package com.jacuum.web.dto

data class MapSnapshot(
    val width: Int,
    val height: Int,
    val startX: Int,
    val startY: Int,
    val totalFloor: Int,
    val tiles: List<TileSnapshot>
) {
    data class TileSnapshot(
        val x: Int, val y: Int,
        val wallNorth: Boolean, val wallSouth: Boolean,
        val wallEast: Boolean, val wallWest: Boolean
    )
}
